File: app/src/main/java/art/coded/wireframe/ui/tab/TabViewModel.java
package art.coded.wireframe.ui.tab;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

import art.coded.wireframe.data.Element;

public class TabViewModel extends ViewModel {

    private final String LOG_TAG = TabViewModel.class.getSimpleName();

    private MutableLiveData<Integer> mIndex = new MutableLiveData<>();
    private LiveData<String> mText = Transformations.map(mIndex, input -> "Hello world from section: " + input);

    public void setIndex(int index) {
        mIndex.setValue(index);
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/ktx/java/art/coded/wireframe/viewmodel/TabViewModel.kt
package art.coded.wireframe.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModelProvider

private val LOG_TAG = TabViewModel::class.java.simpleName

class TabViewModel : ViewModel() {

    private val index = MutableLiveData<Int>()
    val text = Transformations.map(index) { input: Int -> "Hello world from section: $input" }

    fun setIndex(index: Int) {
        this.index.value = index
    }
}

class TabViewModelFactory() : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return TabViewModel() as T
    }
}